core/circuit.go

```diff
@@ -0,0 +1,262 @@
+package core
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+var gridMeterUsed bool // indicates gridmeter is used already fo a circuit to avoid > 1 usage
+
+type Circuit struct {
+	Log    *util.Logger
+	uiChan chan<- util.Param
+
+	Name       string    `mapstructure:"name"`       // meaningful name
+	MaxCurrent float64   `mapstructure:"maxCurrent"` // the max allowed current of this circuit
+	MeterRef   string    `mapstructure:"meter"`      // Charge meter reference
+	Circuits   []Circuit `mapstructure:"circuits"`   // sub circuits as config reference
+
+	parentCircuit *Circuit         // parent circuit reference
+	meterCurrent  api.MeterCurrent // meter to determine phase current
+	vMeter        *VMeter          // virtual meter if no real meter is used
```

Comment:
Why is a separate vMeter required instead of the phaseMeter?

Reply:
phaseMeter bei physischen Meters welche auch Phasenströme liefern. vMeter liefert nur einen Maximalstrom auf Basis aller enthaltenen LP (bzw Consumers).